### PR TITLE
feat: create ito ui

### PR DESCRIPTION
**Pull request by lelenei**
Restore the ITO creation dialog UI

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "lib": ["DOM", "DOM.Iterable", "ES2020"] /* Specify library files to be included in the compilation. */,
     "allowJs": true /* Allow javascript files to be compiled. */,
     // "checkJs": true,                       /* Report errors in .js files. */
-    "jsx": "react-jsx" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
+    "jsx": "react" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
```
Comment (septs) [inline tsconfig.json:10]: why?

Comment (lelenei) [reply tsconfig.json:10]: react-jsx added by vscode

File: package.json [hunk 2]
```diff
@@ -48,6 +48,7 @@
     ]
   },
   "dependencies": {
+    "@date-io/date-fns": "^1.3.13",
```
[the remark this answers is not in the document]
Comment (lelenei) [reply package.json:51]: used datetimepickers component

File: packages/maskbook/src/plugins/ITO/UI/CreateItoGuide.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        },
+    )
+
+    // open the transaction dialog
+    useEffect(() => {
+        if (!poolSettings?.token || createState.type === TransactionStateType.UNKNOWN) return
+        setTransactionDialogOpen({
+            open: true,
+            state: createState,
+            summary: `Create ITO with ${formatBalance(
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CreateItoGuide.tsx:118]: i18n

File: packages/maskbook/src/plugins/ITO/UI/ExchangeTokenPanel.tsx [hunk 4]
```diff
@@ -146,114 +153,97 @@ export function ExchangeTokenPanelUI(props: ExchangetokenPanelUIProps) {
...
 
     useEffect(() => {
-        if (exchangeTokenArray?.length > 0) {
-            const arr = exchangeTokenArray.map((item) => [item.amount, item.token])
-            onChange(arr)
-        }
-    }, [exchangeTokenArray, onChange])
+        onTokenAmountChange(exchangeTokenArray)
+        const addresses = exchangeTokenArray
+            .filter((item) => item && item.token)
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ExchangeTokenPanel.tsx:214]:
```suggestion
            .filter((item) => item?.token)
```

**Review by septs — CHANGES_REQUESTED** [review]

File: packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx [hunk 5]
```diff
@@ -1,66 +1,85 @@
...
     const { t } = useI18N()
-    const classes = useStylesExtends(useStyles(), props)
 
-    const onCreateOrSelect = useCallback(
+    const onCreatePayload = useCallback(
         (payload: ITO_JSONPayload) => {
-            editActivatedPostMetadata((next) => (payload ? next.set(ITO_MetaKey, payload) : next.delete(ITO_MetaKey)))
-            props.onConfirm(payload)
+            // An ITO packet offering 1000000000 ETH for ordering 1000000000 * 500 MAK from public domain
+            /*
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx:27]: remove it.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (lelenei) [reply package.json]: I have restored the package-lock.json and package.json

File: packages/maskbook/src/plugins/ITO/UI/ExchangeTokenPanel.tsx (hunk 4 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/ExchangeTokenPanel.tsx:214]: ok

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx [hunk 6]
```diff
@@ -21,124 +19,110 @@ const useStyles = makeStyles((theme) =>
...
-                    </Grid>
-                    <Grid item xs={6}>
-                        <Paper className={classes.data}>{new Date(itoSettings.startTime).toLocaleString()}</Paper>
-                    </Grid>
+                <Grid item xs={6}>
+                    <Paper className={classes.label}>{t('plugin_ito_begin_times')}</Paper>
+                </Grid>
+                <Grid item xs={6}>
+                    <Paper className={classes.data}>
+                        {poolSettings?.startTime.toLocaleString().replace(/:00$/g, '')}
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: ?????

File: packages/maskbook/src/plugins/ITO/UI/CreateItoGuide.tsx (hunk 3 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateItoGuide.tsx:117]: ok

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx [hunk 7]
```diff
@@ -21,124 +19,110 @@ const useStyles = makeStyles((theme) =>
...
+                    <Paper className={classes.label}>{t('plugin_ito_end_times')}</Paper>
+                </Grid>
+                <Grid item xs={6}>
+                    <Paper className={classes.data}>
+                        {poolSettings?.endTime.toLocaleString().replace(/:00$/g, '')}
+                    </Paper>
+                </Grid>
+                <Grid item xs={12}>
+                    <Typography variant="h5" className={classes.title} component="p">
+                        You can select Existing in ITO to view the selection after successful sending
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:112]: i18n

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx [hunk 8]
```diff
@@ -21,124 +19,110 @@ const useStyles = makeStyles((theme) =>
...
+                    </Typography>
+                </Grid>
+                <Grid item xs={6} className={classes.button}>
+                    <ActionButton fullWidth variant="contained" onClick={onBack}>
+                        {t('plugin_ito_back')}
+                    </ActionButton>
+                </Grid>
+                <Grid item xs={6} className={classes.button}>
+                    <ActionButton fullWidth variant="contained" onClick={onDone}>
+                        {` Send ${poolSettings?.total} ${poolSettings?.token?.symbol} `}
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:122]: i18n

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx (hunk 6 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: remove second

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx (hunk 8 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:122]: ok

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx (hunk 7 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:112]: ok

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx (hunk 6 above)
Comment (septs) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: Date.prototype.toLocaleString()

Result is follow browser first-language.

Comment (septs) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: plz use `Intl.DateTimeFormat` or `date-fns/format`.

Comment (septs) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: CC @Jack-Works any idea?

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 9]
```diff
@@ -110,94 +95,156 @@ export function CreateForm(props: CreateFormProps) {
...
-    }, [])
 
-    const onDaysChange = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
-        const days_ = ev.currentTarget.value.replace(/[,\.]/g, '')
-        console.log(days_)
-        if (days_ === '') setDays(0)
-        else if (/^[0-9]+\d*$/.test(days_)) {
-            setDays(Number.parseInt(days_, 10))
+        if (new BigNumber(tokenAndAmount?.amount ?? '0').isGreaterThan(new BigNumber(tokenBalance)))
+            return `Insufficient ${tokenAndAmount?.token?.symbol} balance`
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:166]: i18n

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 10]
```diff
@@ -110,94 +95,156 @@ export function CreateForm(props: CreateFormProps) {
...
+    ])
+
+    const handleStartTime = useCallback(
+        (timeString) => {
+            const now = new Date()
+            if (endTime === '' && new Date(timeString).getTime() >= now.getTime()) {
+                setStartTime(timeString)
+            }
+            if (
+                new Date(timeString).getTime() < new Date(endTime).getTime() &&
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:196]: why not store `new Date(timeString).getTime()` result?

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 11]
```diff
@@ -110,94 +95,156 @@ export function CreateForm(props: CreateFormProps) {
...
+        tokenAndAmount?.amount,
+        tokenAndAmount?.token?.symbol,
+        tokenAndAmounts,
+        tokenBalance,
+        totalOfPerWallet,
+    ])
+
+    const handleStartTime = useCallback(
+        (timeString) => {
+            const now = new Date()
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:191]: why not `Date.now()`?

File: packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx (hunk 6 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/ConfirmDialog.tsx:98]: ok

File: packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx [hunk 12]
```diff
@@ -1,66 +1,85 @@
...
 import type { ITO_JSONPayload } from '../types'
-import { useStylesExtends } from '../../../components/custom-ui-helper'
+import type { AbstractTabProps } from '../../../extension/options-page/DashboardComponents/AbstractTab'
+import AbstractTab from '../../../extension/options-page/DashboardComponents/AbstractTab'
 import { useI18N } from '../../../utils/i18n-next-ui'
-import AbstractTab, { AbstractTabProps } from '../../../extension/options-page/DashboardComponents/AbstractTab'
-import { editActivatedPostMetadata } from '../../../social-network/ui'
-import { TestForm } from './TestForm'
+import { CreateItoGuide } from './CreateItoGuide'
+import React from 'react'
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx:12]: remove it.

Comment (Jack-Works) [reply packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx:12]: We're using React 17 new JSX transform so you don't need to import React

Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx]: it removed

File: packages/maskbook/src/plugins/ITO/constants.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    ITO_CONTRACT_ADDRESS: {
+        [ChainId.Mainnet]: '',
+        [ChainId.Ropsten]: '',
+        [ChainId.Rinkeby]: '0x52ceb31d6c197b5c039786fbefd6a82df70fdfd6',
+        [ChainId.Kovan]: '',
+        [ChainId.Gorli]: '',
+    },
+}
+
+export const ITO_CONTRACT_BASE_DATE = new Date(1606780800000)
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/constants.ts:18]:
```suggestion
export const ITO_CONTRACT_BASE_DATE = new Date('2020-12-01T00:00:00.000Z').getTime()
```

Comment (septs) [reply packages/maskbook/src/plugins/ITO/constants.ts:18]: providing an helper function

e.q:
```ts
export const now = () => Date.now() - ITO_CONTRACT_BASE_DATE
```

File: packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx [hunk 14]
```diff
@@ -0,0 +1,67 @@
+export const EthIcon = ({ size = 24 }: { size?: number }) => (
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx:1]: plz use https://jakearchibald.github.io/svgomg/ compress here images.

Comment (septs) [reply packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx:1]: plz use css control size.

remove `height` and `width` attribute.

File: packages/maskbook/src/plugins/ITO/helpers.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+import type { ITO_JSONPayload } from './types'
+import schema from './schema.json'
+
+export const ITO_MetadataReader = createTypedMessageMetadataReader<ITO_JSONPayload>(ITO_MetaKey, schema)
+export const renderWithITO_Metadata = createRenderWithMetadata(ITO_MetadataReader)
+
+/**
+ * The greatest common divisor
+ */
+export function gcd(a: BigNumber, b: BigNumber) {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/helpers.ts:13]: What is this algr used for?

File: packages/maskbook/src/plugins/ITO/hooks/usePoolState.ts [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+          type: PoolActionType.UPDATE_BUY_TOKEN
+          token: PartialRequired<EtherTokenDetailed | ERC20TokenDetailed, 'address'>
+      }
+
+function reducer(state: PoolState, action: PoolAction): PoolState {
+    switch (action.type) {
+        case PoolActionType.UPDATE_SELL_TOKEN:
+            return {
+                ...state,
+                sellToken: {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/hooks/usePoolState.ts:41]: you can use `immer` (in our deps) to simplify this code

File: packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+    <svg width={size} height={size} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx:2]: Please use https://material-ui.com/api/svg-icon/#svgicon-api to wrap your icons

File: packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+                fill="#1BA27A"
+            />
+            <path d="M17.628 6.04785H6.26398V8.79585H10.584V12.8279H13.32V8.79585H17.628V6.04785Z" fill="white" />
+            <path
+                d="M19.224 12.204C19.224 11.304 15.984 10.584 11.976 10.584C7.96797 10.584 4.72797 11.316 4.72797 12.204C4.72797 12.996 7.24797 13.656 10.584 13.8V19.584H13.32V13.8C16.68 13.656 19.224 12.996 19.224 12.204ZM11.976 13.26C8.41197 13.26 5.51997 12.696 5.51997 12C5.51997 11.304 8.41197 10.74 11.976 10.74C15.54 10.74 18.432 11.304 18.432 12C18.432 12.696 15.54 13.26 11.976 13.26Z"
+                fill="white"
+            />
+        </g>
+        <defs>
+            <clipPath id="clip0">
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx:62]: Will it conflict with other SVGs?

File: packages/maskbook/src/plugins/ITO/assets/background.ts [hunk 19]
```diff
@@ -0,0 +1 @@
...TR5WwnOipmWswUWcqZqgtYIsEjVnvQev7C1Hz9WKTwSW5RR7vrhs3/98Pnyb5/7/v+8/9Pf+4PXGX98l6h2gmqpWqzNqB/Xe2gFGsQj73rhsYgtODT+3guyBuUeB3cm2WZuJQL4BC359WeSrYUgrNbb7EAj7eLHEyhljQWZHQ3uST8Llxznl1IFb9v/9dUfH4/la/WT5bsflcNXfv6fv7uWq6/dP5O1kuZnxVZg5LQo76EVXh+sZ2L1SMkZQVKPEZkWaRYyJx2r9dICl4jhV5QkZKTRYIsJf6b15iH/mdatdXEx22c+U6b24iGpdtHtP6d1KetXq1VbP1m+WN5/93V2P/WF8pd/ejhcf7WUd9b7pFsJ10KekcApT5Qx/CxbEOfWZLs3ZJPRY0amTA7QfkQDo3hZe1jE7NF606angVJ4r7DmGsvc5fE774WcH9LYvYSMfFiv1uMff/s7H34dn1x966MfvKw/v/2b5ennP3n+7e+dPvh4LU9+7XXid88rWTsb3j6iqQQ6a2u7QrOSpDzLdejoV4ZrUD/LNRUnRzmWa2gUJKK+o7DWuQVI/REZnuZVkTFwQTwZeWW1VVZefOGK9zQDUqAgFrwjYJxp282a3vNcPM8xOz2dhyrqfWifo7b+eK6W/mEZe5a+Z50LLc/1hpThmcs8czQQnResPPSjEj9+bdH+0Zd+7m/+7NnzFy/w+dXvf/m98sEv/mz5i2/+782z/3zx4jd+6V/+6eb0/jderZ//3OuG+dU3l5OaYiWmqQBu0kq4sB49g3Td4BpcZ7nGOunSQwuiZGsl+xGyyBbYG9nukbipBUTJlm7/KeU8gEr7jGaTm6W/WOcL9G1r/ppngrQVVL4fwVIfa1+zTv7WvLPJtqajnOEZC5ExFBnLZmPsdWHr352O5Xe+cPjGN1/9/9X1R//9g5f4so7a98qtTvxp/eCv/qH+/ttnv/fL7//hf1195U8+Xd//rdN6+N21HL98KOt7t5fdkMs1FcYKzkIa2EP7xHBNfZivir0z3jiu8XQALalnTPJS2W9DXhltl9n+sDqjeVJClSRNXoY1SErbl6EgaccLxqTFwpotJWufCXdP1ft+OrjGuyVJS/4eP+yM+eShYO69r2V9/rqE54dD+dbN98vXnz3/8JP6+Zc++Jn65dkg+SH9kyDM9k455gAAAABJRU5ErkJggg==`
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/assets/background.ts:1]: Why not a seperate PNG file?

File: packages/maskbook/src/plugins/ITO/assets/ItoLabelIcon.tsx [hunk 20]
```diff
@@ -0,0 +1,17 @@
+export const ItoLabelIcon = ({ size = 24 }: { size?: number }) => (
+    <svg width={size} height={size} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/assets/ItoLabelIcon.tsx:2]: Review in tokenIcon.tsx also applies here

File: packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx (hunk 17 above)
Comment (guanbinrui) [reply packages/maskbook/src/plugins/ITO/assets/tokenIcon.tsx:2]: In this case, `<TokenIcon />` is enough.

File: packages/maskbook/src/plugins/ITO/api/useExchangeTokenAmountstate.ts [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+      }
+function reducer(
+    state: ExchangeTokenAndAmountState[],
+    action: ExchangeTokenAndAmountAction,
+): ExchangeTokenAndAmountState[] {
+    switch (action.type) {
+        case ExchangeTokenAndAmountActionType.ADD:
+            return [
+                ...state,
+                {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/api/useExchangeTokenAmountstate.ts:49]: 🤔

File: packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+    )
+
+    const state = useState(0)
+
+    const tabProps: AbstractTabProps = {
+        tabs: [
+            {
+                label: t('plugin_ito_create_new'),
+                children: <CreateItoGuide onCreate={onCreatePayload} />,
+                p: 0,
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/UI/CompositionDialog.tsx:70]: This property has been changed to `sx: {p:0}`

**Comment by guanbinrui** [comment]
This PR includes too many stuff that should not belong to it. Many contents duplicated with  #1978. @lelenei  Please pick your new commits on top of #1978 and force-push to your own branch again. The duplicated part will vanish.

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 23]
```diff
@@ -0,0 +1,325 @@
...
+            padding: 12,
+        },
+        date: {
+            flex: 1,
+            padding: theme.spacing(1),
+        },
+    }),
+)
+
+export interface CreateFormProps extends withClasses<KeysInferFromUseStyles<typeof useStyles>> {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:57]: `KeysInferFromUseStyles` has been deprecated. Please don't use it. See the comment on the definition on the `KeysInferFromUseStyles` for details

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 24]
```diff
@@ -0,0 +1,325 @@
...
+export function CreateForm(props: CreateFormProps) {
+    const { onChangePoolSettings, onNext, onConnectWallet } = props
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), props)
+
+    const account = useAccount()
+    const chainIdValid = useChainIdValid()
+
+    const [message, setMessage] = useState('')
+    const [totalOfPerWallet, setTotalOfPerWallet] = useState('1')
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:72]: Why not use number?

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 25]
```diff
@@ -0,0 +1,325 @@
...
+    const [tokenAndAmounts, setTokenAndAmounts] = useState<ExchangeTokenAndAmountState[]>([])
+    const [tokenAndAmount, setTokenAndAmount] = useState<ExchangeTokenAndAmountState>()
+    const [startTime, setStartTime] = useState('')
+    const [endTime, setEndTime] = useState('')
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:76]: Why not use `Date`?

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx [hunk 26]
```diff
@@ -0,0 +1,325 @@
...
+        onChangePoolSettings(poolSettings)
+    }, [onChangePoolSettings, poolSettings])
+
+    // balance
+    const { value: tokenBalance = '0', loading: loadingTokenBalance } = useTokenBalance(
+        tokenAndAmount?.token?.type ?? EthereumTokenType.Ether,
+        tokenAndAmount?.token?.address ?? '',
+    )
+
+    const validationMessage = useMemo(() => {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:147]: I suggest not using useMemo here

Comment (Jack-Works) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:147]: or split them into a samller chunks

File: packages/maskbook/src/plugins/ITO/hooks/usePoolState.ts (hunk 16 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/hooks/usePoolState.ts:41]: ok

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx (hunk 10 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx]: Limit the range of date input

File: packages/maskbook/src/plugins/ITO/constants.ts (hunk 13 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/constants.ts:18]: Rebase introduced

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx (hunk 24 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:72]: TextField need

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx (hunk 23 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:57]: ok

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx (hunk 25 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:76]: Limit the range of date input

File: packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx (hunk 26 above)
Comment (lelenei) [reply packages/maskbook/src/plugins/ITO/UI/CreateForm.tsx:147]: ok

**Pull request closed without merging**